azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/PromptWrapper.java

```diff
@@ -0,0 +1,276 @@
+import org.apache.maven.plugin.logging.Log;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluationException;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluator;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class PromptWrapper {
+    private ExpressionEvaluator expressionEvaluator;
+    private IPrompter prompt;
+    private Map<String, Map<String, Object>> templates;
+    private Map<String, Object> commonVariables;
+    private SchemaValidator validator;
+    private Log log;
+
+    public PromptWrapper(ExpressionEvaluator expressionEvaluator, Log log) {
+        this.expressionEvaluator = expressionEvaluator;
+        this.log = log;
+    }
+
+    public void initialize() throws IOException {
+        prompt = new DefaultPrompter();
+        validator = new SchemaValidator();
+        templates = new HashMap<>();
+        commonVariables = new HashMap<>();
+        final Yaml yaml = new Yaml();
+        try (final InputStream inputStream = this.getClass().getResourceAsStream("/MessageTemplates.yaml")) {
+            final Iterable<Object> rules = yaml.loadAll(inputStream);
+
+            for (final Object rule : rules) {
+                final Map<String, Object> map = (Map<String, Object>) rule;
+                templates.put((String) map.get("id"), map);
+            }
+        }
+    }
+
+    public void putCommonVariable(String key, Object obj) {
+        this.commonVariables.put(key, obj);
+    }
+
+    public <T> T handleSelectOne(String templateId, List<T> options, T defaultEntity, Function<T, String> getNameFunc)
+            throws IOException, SpringConfigurationException, NoResourcesAvailableException {
+        final Map<String, Object> variables = createVariableTables(templateId);
+        final boolean isRequired = TemplateUtils.evalBoolean("required", variables);
+        if (options.size() == 0) {
+            if (isRequired) {
+                throw new NoResourcesAvailableException(TemplateUtils.evalText("message.empty_options", variables));
+            }
+            final String warningMessage = TemplateUtils.evalText("message.empty_options", variables);
+            if (StringUtils.isNotBlank(warningMessage)) {
+                log.warn(warningMessage);
+            }
+            return null;
+        }
+        final boolean autoSelect = TemplateUtils.evalBoolean("auto_select", variables);
+        if (options.size() == 1) {
+            if (autoSelect) {
+                log.info(TemplateUtils.evalText("message.auto_select", variables));
+                return options.get(0);
+            }
+            if (!this.prompt.promoteYesNo(TemplateUtils.evalText("promote.one", variables),
```

Comment:
Why not just select the one if the option is required?

Reply:
If autoSelect is false, then we cannot take the only one choice.